src/libYARP_dev/src/devices/msgs/ros/include/TickTime.h

```diff
@@ -23,6 +24,39 @@ class TickTime : public yarp::os::idl::WirePortable
     {
     }
 
+    TickTime(double timestamp) :
+            sec(0),
+            nsec(0)
+    {
+        uint64_t time = (uint64_t) (timestamp * 1000000000UL);
+        sec = (time / 1000000000UL);
+        nsec = (time % 1000000000UL);
+        if (sec > UINT_MAX) {
```

Comment:
`UINT_MAX` is the platform-dependent maximum value for `unsigned int`, so I do not understand the reason of this check (comparing against platform-dependent limits?).
Furthermore, consider that in original ros code and in the remove function, `sec` and `nsec` were `uint64_t`, and were casted to 32-bit quantities only after this check.

Reply:
Uh, I see... the check was done before the cast... sorry for the mess

Reply:
Regarding the `UINT_MAX`, I think it should be `UINT32_MAX`, at least for us as we depend on C++11 . I opened an issue upstream https://github.com/ros/roscpp_core/issues/74 to discuss about this.

Reply:
I did some changes, restored the `uint64_t`, changed `UINT_MAX` to `UINT32_MAX`...
Please have a look at the changes...

BTW, they have only 74 issues on roscpp_core? :thinking:

Reply:
Note that roscpp_core is not ros_comm (https://github.com/ros/ros_comm).